sound/soc/sof/control.c

```diff
@@ -385,8 +385,9 @@ int snd_sof_bytes_ext_volatile_get(struct snd_kcontrol *kcontrol, unsigned int _
 
 	/* check data size doesn't exceed max coming from topology */
 	if (cdata->data->size > be->max - sizeof(const struct sof_abi_hdr)) {
-		dev_err_ratelimited(scomp->dev, "error: user data size %d exceeds max size %lu.\n",
-				    cdata->data->size, be->max - sizeof(const struct sof_abi_hdr));
+		dev_err_ratelimited(scomp->dev, "error: user data size %d exceeds max size %zu.\n",
+				    cdata->data->size,
+				    (size_t)be->max - sizeof(const struct sof_abi_hdr));
```

Comment:
mmh, type-casts are evil(TM). Is there any specific reason for this one? It's only to fix the `printk()` format, right? The calculation, the value would be exactly the same if we just used `%d`? Maybe just do that then... Or is this because `size_t` is a different type on 32-bit builds, ARM?

Reply:
be->max is an int, but after arithmetic with sizeof(), it's a "unsigned int" right. so would "%u" as printf syntax solve ths issue without a cast?

Reply:
@lyakh yes size_t is different. 
I tried earlier with a %d but it wouldn't work for x86_64. the %lu fails on 32-bit. 
So I settled on a typecast to force the use of size_t, and let the compiler adjust.
If you have a better solution I am all ears, just trying to avoid an error reported upstream.

Reply:
> be->max is an int, but after arithmetic with sizeof(), it's a "unsigned int" right. so would "%u" as printf syntax solve ths issue without a cast?

@kv2019i no, sizeof() returns a size_t, which can be 32 or 64 bits, so that impacts the results.

Reply:
The cast seems unnecessary, `%zd` should be enough. Have you tried with `%zd` only and no cast?

I think we can safely assume `size_t` has a rank always greater or equal to `int`.

In user space the following code compiles with both `-m32` and `-m64` without any warning:
```C
  int i = -1;
  size_t s = 42;
  printf ("%zd", i - s);
```

https://wiki.sei.cmu.edu/confluence/display/c/INT02-C.+Understand+integer+conversion+rules
> Usual arithmetic conversions
> 3. If the operand that has unsigned integer type has rank greater than or equal to the rank of the type of the other operand, the operand with signed integer type is converted to the type of the operand with unsigned integer type.

Reply:
i'll give it a try without the cast.

Reply:
Doesn't work without the cast:

https://travis-ci.org/github/thesofproject/linux/jobs/722150721

````
sound/soc/sof/topology.c:1165:23: error: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 3 has type ‘size_t {aka unsigned int}’ [-Werror=format=]
````

going back to the initial proposal with the cast...

Reply:
That's because you switch to `%lu` (by accident?). I meant no cast and `%zu`.

Reply:
see https://github.com/thesofproject/linux/commit/e1cc3448877ef8a4786326620a37c01287bc54c3, didn't work.

Reply:
I see that e1cc344 has `%lu` and not the `%zu` that I suggested.

Reply:
it's %zu alright https://github.com/thesofproject/linux/commit/e1cc3448877ef8a4786326620a37c01287bc54c3#diff-6782c454986b1c6b9d4916315dd9dec7R391

Reply:
My bad, I was misled by the error message in topology.c which you hadn't changed yet.